frontend/database/modules/datasets.js

```diff
@@ -479,6 +486,22 @@ async function _updatePagination({ id, size, page }) {
   return pagination;
 }
 
+async function _chooseContinueOrCancel(action, dataset, query, sort, size) {
+  return await Notification.dispatch("notify", {
+    message: "Pending actions will be lost when the page is refreshed",
+    type: "warning",
+    numberOfChars: 20000,
+    buttonText: "Ok, got it!",
+    async onClick() {
+      await action(dataset, query, sort, size);
+      await _updateViewSettings({
+        id: dataset.name,
+        data: { arePendingRecords: false },
+      });
+    },
+  });
+}
```

Comment:
code cleanning : one thing by line => create the object before the call for notification toast